Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp

```diff
@@ -80,6 +80,10 @@ static RetainPtr<CFMutableDictionaryRef> createImageSourceOptions()
         CFDictionarySetValue(options.get(), kCGImageSourceEnableRestrictedDecoding, kCFBooleanTrue);
 #endif
 
+#if HAVE(IMAGE_RESTRICTED_DECODING) && HAVE(AVIF)
+    CFDictionarySetValue(options.get(), kCGImageSourceEnableRestrictedDecoding, kCFBooleanTrue);
```

Comment:
I'm a bit confused by this. This line doesn't seem to be AVIF-specific, but it's behind `HAVE(AVIF)`.

Does this line have repercussions beyond just supporting AVIF?

Reply:
You are right. I deleted this new code and I changed the code above it to enable the restricted (software) image decoding if HEIF or AVIF are enabled in WebKit.